src/main/java/cs2103/v15_1j/jimjim/searcher/JJSearcher.java

```diff
@@ -0,0 +1,80 @@
+package cs2103.v15_1j.jimjim.searcher;
+import java.util.List;
+
+import cs2103.v15_1j.jimjim.model.DataLists;
+import cs2103.v15_1j.jimjim.model.DeadlineTask;
+import cs2103.v15_1j.jimjim.model.Event;
+import cs2103.v15_1j.jimjim.model.FloatingTask;
+
+public class JJSearcher implements Searcher {
+	@Override
+	public DataLists search(List<Filter> filters, DataLists masterLists) {
+		DataLists resultsList = new DataLists(); // Result to be returned
+		
+		List<DeadlineTask> masterDeadlineTasksList = masterLists.getDeadlineTasksList();
+		List<FloatingTask> masterFloatingTasksList = masterLists.getFloatingTasksList();
+		List<Event> masterEventsList = masterLists.getEventsList();
+		
+		List<DeadlineTask> searchResultsDeadlineTasksList = resultsList.getDeadlineTasksList();
+		getDeadlineTasksSearchResults(filters, masterDeadlineTasksList, searchResultsDeadlineTasksList);
```

Comment:
I don't like this way of programming. Create too many mutable objects which may lead to errors. I think it's better if `get...` return a `List<>`. you save those lists and later return a newly constructed `DataLists` from these `List<>`s.
Same comment for the other functions.